### ISSUE TITLE
InfoBar: Focus leaves InfoBar when action button is removed

### ISSUE THREAD

**Issue by gabbybilka** [microsoft/microsoft-ui-xaml#4531]
**Describe the bug**
Keyboard focus jumps outside of the InfoBar if an action button is removed from the InfoBar while it has focus.

**Steps to reproduce the bug**
1. Create an InfoBar with an action and close button.
2. Keyboard focus on the action button
3. Remove the action button in code behind
  - Current behavior: Keyboard focus jumps to the next focusable item outside the InfoBar.
  - Expected behavior: Keyboard focus moves to the next available focusable item, the close button.

NuGet package version: 
[Microsoft.UI.Xaml 2.6 prerelease]

**Open questions**
- What should the behavior be if there are no focusable elements in the InfoBar after a focusable element is removed?
- Should the entire InfoBar be focusable to mitigate this issue, even though it can't be pressed?
  - This then requires an additional tab to get into the action or close button.

**Comment by gabbybilka** [comment]
@YuliKl as FYI, what would you recommend as the expected behavior here?

**Comment by ranjeshj** [comment]
This seem like an issue with the framework code where the focused item picked after the currently focused element is removed from the tree is incorrect. It can be worked around from the app code which removes the button to put focus on the item that it wants to next.

**Comment by johnpf74** [comment]
Why doesn't focus move to the close button in this case to begin with? The close button is the next focusable element after the action button.

**Comment by RBrid** [comment]
We have identified a bug in the XAML framework. Before we get a chance to address it, you can set the ActionButton's TabIndex to avoid this behavior:  myInfoBar.ActionButton.TabIndex = 1;

**Comment by RBrid** [comment]
I cannot no longer repro this problem in WinUI3 1.4.
